### PR TITLE
Use poetry to generate docs

### PR DESCRIPTION
**Pull request by shubhank-saxena**
@armenzg do let me know if any further changes are required!

### CODE REVIEW

File: docs/common_tasks.md [hunk 1]
```diff
@@ -11,11 +11,12 @@ To use this within the Docker environment, run:
...
 
-Or if you would rather not use Docker, instead activate a virtualenv on the host
-machine, and from the root of the Treeherder repo, run:
+Or if you would rather not use Docker, instead use poetry,run:
 
 ```bash
-> pip install -r requirements/docs.txt
+> pip install poetry
+> poetry install
+> poetry shell
```
Comment (armenzg) [inline docs/common_tasks.md]:
```suggestion
> poetry run mkdocs serve
```

Comment (shubhank-saxena) [reply docs/common_tasks.md]: That's strange. I did try this first. Wasn't working on my machine. Maybe some issues. Let me give it a try and will do the changes!

File: .gitignore [hunk 2]
```diff
@@ -28,3 +28,6 @@ celerybeat-schedule
 coverage
 coverage.xml
 .coverage
+
+# Ignore poetry lock
+poetry.lock
```
Comment (armenzg) [inline .gitignore]: Why are you ignoring `poetry.lock`?

File: docs/common_tasks.md [hunk 3]
```diff
@@ -11,12 +11,12 @@ To use this within the Docker environment, run:
 docker-compose run -p 8000:8000 backend mkdocs serve -a 0.0.0.0:8000
 ```
 
-Or if you would rather not use Docker, instead activate a virtualenv on the host
-machine, and from the root of the Treeherder repo, run:
+Or if you would rather not use Docker, instead use poetry,run:
 
 ```bash
```
Comment (armenzg) [inline docs/common_tasks.md]: Switch to `console`.

File: docs/common_tasks.md [hunk 4]
```diff
@@ -11,12 +11,12 @@ To use this within the Docker environment, run:
...
-> pip install -r requirements/docs.txt
-> mkdocs serve
+> pip install poetry
+> poetry install
+> poetry run mkdocs serve
```
Comment (armenzg) [inline docs/common_tasks.md]: Add a note outside of this block saying:

NOTE: On Windows you might need to fallback `python -m venv venv` or `virtualenv` to manage your virtualenv if `poetry` does not work for you.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+[tool.poetry]
```
Comment (armenzg) [inline pyproject.toml]: Move this file and `poetry.lock` under `docs/`

**Pull request merged**